src/url_components.cpp

```diff
@@ -6,40 +6,52 @@
 
 namespace ada {
 
-  std::pair<bool,uint32_t> url_components::check_offset_consistency() const noexcept {
+  bool url_components::check_offset_consistency() const noexcept {
+   /**
+    * https://user:pass@example.com:1234/foo/bar?baz#quux
+    *       |     |    |          | ^^^^|       |   |
+    *       |     |    |          | |   |       |   `----- hash_start
+    *       |     |    |          | |   |       `--------- search_start
+    *       |     |    |          | |   `----------------- pathname_start
+    *       |     |    |          | `--------------------- port
+    *       |     |    |          `----------------------- host_end
+    *       |     |    `---------------------------------- host_start
+    *       |     `--------------------------------------- username_end
+    *       `--------------------------------------------- protocol_end
+    */
     // These conditions can be made more strict.
     uint32_t index = 0;
     if(protocol_end != url_components::omitted) {
-      if(protocol_end < index) { return {false,0}; }
+      if(protocol_end < index) { return false; }
       index = protocol_end;
     }
     if(username_end != url_components::omitted) {
-      if(username_end < index) { return {false,0}; }
+      if(username_end < index) { return false; }
       index = username_end;
     }
     if(host_start != url_components::omitted) {
-      if(host_start < index) { return {false,0}; }
+      if(host_start < index) { return false; }
       index = host_start;
     }
     if(port != url_components::omitted) {
-      if(port > 0xffff) { return {false,0}; }
+      if(port > 0xffff) { return false; }
       uint32_t port_length = helpers::fast_digit_count(port) + 1;
-      if(index + port_length < index) { return {false,0}; }
+      if(index + port_length < index) { return false; }
       index += port_length;
     }
     if(pathname_start != url_components::omitted) {
-      if(pathname_start < index) { return {false,0}; }
+      if(pathname_start < index) { return false; }
```

Comment:
there are some fields like pathname that can't be url_components::omitted, but i'll fix it with a follow up pr.